apps/opencs/model/world/collection.hpp

```diff
@@ -120,6 +120,8 @@ namespace CSMWorld
 
             virtual const Record<ESXRecordT>& getRecord (int index) const;
 
+            virtual Record<ESXRecordT>& getRecord (int index);
```

Comment:
Workaround is not co pretty.  Both getRecord() and setRecord() provide access to const references only, so this messing around with assign() is needed.  It would be useful to have something like modifyRecord() with a non-const reference.

    template<typename ESXRecordT, typename IdAccessorT>
    void SubCellCollection<ESXRecordT, IdAccessorT>::addNestedRow(int row, int column, int position)
    {
        Record<ESXRecordT> record;
        record.assign(Collection<ESXRecordT, IdAccessorT>::getRecord(row));

        getAdapter(Collection<ESXRecordT, IdAccessorT>::getColumn(column)).addNestedRow(record, position);

        Collection<ESXRecordT, IdAccessorT>::setRecord(row, record);
    }